.github/workflows/reusable-change-detection.yml

```
@@ -83,7 +83,10 @@ jobs:
           # into the PR branch anyway.
           #
           # https://github.com/python/core-workflow/issues/373
-          git diff --name-only "origin/$GITHUB_BASE_REF.." | grep -qvE '(\.rst$|^Doc|^Misc|^\.pre-commit-config\.yaml$|\.ruff\.toml$|\.md$|mypy\.ini$)' && echo "run-tests=true" >> "$GITHUB_OUTPUT" || true
+          git diff --name-only "origin/$GITHUB_BASE_REF.." | grep -qvE    \
+              -e '(\.rst$|\.md$|^Doc|^Misc|^\.github/CODEOWNERS$)'        \
+              -e '(^\.pre-commit-config\.yaml$|\.ruff\.toml$|mypy\.ini$)' \
```

These regexes are dense and a bit tricky to read/update, shall we split them out to make each individual file pattern easier to check on its own? (I've not tested this)
```suggestion
          git diff --name-only "origin/$GITHUB_BASE_REF.." | grep -qvE \
              -e '\.md$' \
              -e '\.rst$' \
              -e '\.ruff\.toml$' \
              -e '^\.github/CODEOWNERS$' \
              -e '^\.pre-commit-config\.yaml$' \
              -e '^Doc' \
              -e '^Misc' \
              -e 'mypy\.ini$' \
```

---

I think it would be clearer. According to the manpage:

```text
-e PATTERN, --regexp=PATTERN
      Use  PATTERN  as  the  pattern.   If  this option is used multiple times or is combined with the -f (--file) option, search for all
      patterns given.  This option can be used to protect a pattern beginning with “-”.
```

So it's equivalent to use the `|`. I would group the patterns as follows: first directories, then config files (pre-commit etc) and the arbitrary files with extensions (or we can put extensions first). I don't know if grep optimizes the patterns from the most likely to the less likely for instance (I know that `find` is doing some pretty awesome stuff when it comes to predicates though)